planner/core/expression_rewriter.go

```diff
@@ -157,6 +157,7 @@ func (b *PlanBuilder) getExpressionRewriter(ctx context.Context, p LogicalPlan)
 	rewriter.ctxStack = rewriter.ctxStack[:0]
 	rewriter.ctxNameStk = rewriter.ctxNameStk[:0]
 	rewriter.ctx = ctx
+	rewriter.allNames = b.allNames
```

Comment:
Can we remove the filed `allNames` in `PlanBuilder` and change this line to `rewriter.allNames = p.OutputNames()`?